core/src/shred_fetch_stage.rs

```diff
@@ -86,16 +87,13 @@ impl ShredFetchStage {
                 let shred_type = p.data[OFFSET_OF_SHRED_TYPE];
                 if shred_type == DATA_SHRED || shred_type == CODING_SHRED {
                     // Shred filter
-                    let slot_received =
-                        shreds_received
-                            .entry((slot, shred_type))
-                            .or_insert_with(|| {
-                                BitVec::new_fill(false, MAX_DATA_SHREDS_PER_SLOT as u64)
-                            });
-                    if !slot_received.get(index.into()) {
+                    let mut hasher = blake3::Hasher::new();
+                    hasher.update(&p.data[0..p.meta.size]);
```

Comment:
is there a big perf implication here for computing a cryptographic hash like blake3 for every shred? 

Here we only care about detecting duplicates, so can we use a weaker hash by using something like hashbrown::HashSet (uses ahash): https://github.com/tkaitchuck/aHash

Reply:
Yes, but a potential bad actor can easily find collisions in those and I think blake3 is fairly close in performance. I can test it, but if it's close, I would rather have the safety of blake3.

Reply:
@sakridge we can add a seed, i.e. generate a seed every 400 ms to add to the key. I think purported benchmarks were supposed to be 16-32x faster.

Reply:
blake3 can reject at 650k shreds/s per core, but ahash is faster.